### PR TITLE
Emit render phase events (closes #88)

### LINKED ISSUE #88: Emit events to inidicate when render phases have completed
Something like:
1. `partialrender`
2. `fullrender`

Then GaiaFastList can emit an event out to the app so that the app knows when to reveal the view. Like `firstrender` which could fire as a result of either:

A. Cached content injected
B. `FastList` emitted `partialrender` event

Depending on component configuration and cache state.

### PR DESCRIPTION
**Pull request by wilsonpage**

### CODE REVIEW

**Comment by etiennesegonzac** [comment]
And we'll want tests :)

**Comment by wilsonpage** [comment]
@etiennesegonzac r?

**Comment by wilsonpage** [comment]
We now have two promises as render stage hooks:
- `rendered` (something ready to show)
- `complete` (initialization is fully complete)

I decided to stick with Promises as they're nicer to use and work better for one-time events like this.

File: fast-list-edit.js [hunk 1]
```diff
@@ -161,7 +161,7 @@ function fastListEdit(list) {
 
     if (ctx.moveUp.size === upCount &&
         ctx.moveDown.size === downCount) {
-      return;
+      return debug('poook');
     }
```
Comment (etiennesegonzac) [inline fast-list-edit.js:165]: looks like we don't need those changes here :)

**Comment by etiennesegonzac** [comment]
Can you do a quick rebase, it should be very quick to review after that.

File: fast-list.js [hunk 2]
```diff
@@ -404,7 +412,7 @@ FastList.prototype = {
   },
 
   reloadData: function() {
-    return this.rendered = schedule.mutation(function() {
+    return schedule.mutation(function() {
```
Comment (wilsonpage) [inline fast-list.js:415]: Decided redefining `fastList.rendered` was kinda confusing as promises are meant to be a one time event. This function returns a Promise so user is still able to observe that if they wish to know when `reloadData` has completed.

File: fast-list.js [hunk 3]
```diff
@@ -96,8 +93,8 @@ FastList.prototype = {
    *
    * @private
    */
-  renderPhase1: function() {
-    debug('render phase 1');
+  setupPhase1: function() {
```
Comment (wilsonpage) [inline fast-list.js:96]: 'setup' is a nicer word as it's clearer it is a one time thing and not associated with normal scroll rendering.

File: .jshintrc [hunk 4]
```diff
@@ -22,8 +22,6 @@
   "proto": true,
   "boss": true,
 
-  "esnext": true,
```
Comment (wilsonpage) [inline .jshintrc:25]: Let's actually warn when we break Safari with ES6 syntax :)

**Comment by wilsonpage** [comment]
@etiennesegonzac done :)

File: fast-list.js (hunk 3 above)
Comment (etiennesegonzac) [inline fast-list.js:96]: agreed

**Comment by etiennesegonzac** [comment]
All good!

**Pull request merged**